src/utransport.rs

```diff
@@ -514,33 +505,32 @@ impl UTransport for UPClientZenoh {
                 self.send_request(&zenoh_key, payload, attributes).await
             }
             UMessageType::UMESSAGE_TYPE_RESPONSE => {
-                Validators::Response
+                UAttributesValidators::Response
                     .validator()
                     .validate(&attributes)
                     .map_err(|e| {
-                        UStatus::fail_with_code(
-                            UCode::INVALID_ARGUMENT,
-                            format!("Wrong Response UAttributes {e:?}"),
-                        )
+                        let msg = format!("Wrong Response UAttributes: {e:?}");
+                        log::error!("{msg}");
+                        UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg)
                     })?;
                 // Get Zenoh key
                 let topic = attributes.clone().source;
                 let zenoh_key = UPClientZenoh::to_zenoh_key_string(&topic)?;
                 // Send Response
                 self.send_response(&zenoh_key, payload, attributes).await
             }
-            UMessageType::UMESSAGE_TYPE_UNSPECIFIED => Err(UStatus::fail_with_code(
-                UCode::INVALID_ARGUMENT,
-                "Wrong Message type in UAttributes",
-            )),
+            UMessageType::UMESSAGE_TYPE_UNSPECIFIED => {
+                let msg = "Wrong Message type in UAttributes".to_string();
+                log::error!("{msg}");
+                Err(UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg))
+            }
         }
     }
 
     async fn receive(&self, _topic: UUri) -> Result<UMessage, UStatus> {
-        Err(UStatus::fail_with_code(
-            UCode::UNIMPLEMENTED,
-            "Not implemented",
-        ))
+        let msg = "Not implemented".to_string();
+        log::error!("{msg}");
+        Err(UStatus::fail_with_code(UCode::UNIMPLEMENTED, msg))
     }
 
     async fn register_listener(
```

Comment:
btw -- do we need to have now a `register_notification_listener()`? Or is that handled by one of the existing functions?

Reply:
No. I think we have no way to distinguish publish and notification only from UUri.
We can only handle them with `register_publish_listener` together. (Maybe need a better name?)

Reply:
Okay. I think you're right, some name change may be needed.

Reply:
Done